### PR TITLE
PP 353 implement the main structure for the user journey tracking

### PR DESCRIPTION
**Pull request by ValentinaIancu-Gini**
This is adding the main structure for analytics to be able to start adding analytics events that we need to the user journey story.

### CODE REVIEW

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Tracking/AnalyticsManager.swift [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+class AnalyticsManager {
+    private static let mixPanelToken = "6262hhdfhdb929321222" // this id is fake we need to replace it
+    static var mixpanelInstance: MixpanelInstance?
+
+    static func initializeAnalytics() {
+        mixpanelInstance = Mixpanel.initialize(token: mixPanelToken, trackAutomaticEvents: false)
+        
+        // Identify the user with the deviceID
+        let deviceID = UIDevice.current.identifierForVendor?.uuidString
```
Comment (zladzeyka) [inline CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Tracking/AnalyticsManager.swift:18]: [Clarification]: If I'm not mistaken, we will need to add additional info in our privacy manifest that we've gathered device data. 
Do we need to ask for tracking permissions in that case?

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Tracking/AnalyticsProperties.swift [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    }
+}
+
+extension Int: AnalyticsPropertyValue {
+    func analyticsPropertyValue() -> Int {
+        return self
+    }
+}
+
+extension UInt: AnalyticsPropertyValue {
```
Comment (zladzeyka) [inline CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Tracking/AnalyticsProperties.swift:30]: [Clarification]: Do we plan to pass the properties `UInt` and `Double` if we are not sure - I propose to clean it and add it later when we need it. 😇

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Tracking/AnalyticsScreen.swift [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+//  AnalyticsScreen.swift
+//
+//  Copyright © 2024 Gini GmbH. All rights reserved.
+//
+
+
+import Foundation
+
+enum AnalyticsScreen: String {
+    case none
```
Comment (zladzeyka) [inline CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Tracking/AnalyticsScreen.swift:11]: [Clarification]: when do we plan to use `none`? 🧐

**Review by zladzeyka — CHANGES_REQUESTED** [review]
Great job,
I left some small clarification questions, ping me if more details are needed.😊

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Screen API Coordinator/GiniScreenAPICoordinator+Camera.swift [hunk 4]
```diff
@@ -61,6 +61,8 @@ extension GiniScreenAPICoordinator: CameraViewControllerDelegate {
     }
 
     func cameraDidAppear(_ viewController: CameraViewController) {
+        //TODO: check where to place this best since openWith is starting differently
```
[the remark this answers is not in the document]
Comment (ValentinaIancu-Gini) [reply CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Screen API Coordinator/GiniScreenAPICoordinator+Camera.swift:64]: Yes, I will keep it until I will work on that specific case.

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Tracking/AnalyticsManager.swift (hunk 1 above)
Comment (ValentinaIancu-Gini) [reply CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Tracking/AnalyticsManager.swift:18]: No we don't need to ask for anything. 
"Mixpanel does not use IDFA, so it does not require user permission through the AppTrackingTransparency(ATT) framework."

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Tracking/AnalyticsProperties.swift (hunk 2 above)
Comment (ValentinaIancu-Gini) [reply CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Tracking/AnalyticsProperties.swift:30]: I cleaned them in another pr 🙈 .

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Tracking/AnalyticsScreen.swift (hunk 3 above)
Comment (ValentinaIancu-Gini) [reply CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Tracking/AnalyticsScreen.swift:11]: Theoretically no. I was thinking to have a value as a fallback if we need it.

File: BankSDK/GiniBankSDKExample/GiniBankSDKExample/AppCoordinator.swift [hunk 5]
```diff
@@ -35,6 +35,7 @@ final class AppCoordinator: Coordinator {
         configuration.multipageEnabled = true
         configuration.flashToggleEnabled = true
         configuration.localizedStringsTableName = "LocalizableCustomName"
+        configuration.onboardingShowAtLaunch = false
```
[the remark this answers is not in the document]
Comment (ValentinaIancu-Gini) [reply BankSDK/GiniBankSDKExample/GiniBankSDKExample/AppCoordinator.swift:38]: 😆 It was in the beginning when I tested Onboarding events while investigating which platform to use.
I will remove it.

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Tracking/AnalyticsScreen.swift (hunk 3 above)
Comment (ValentinaIancu-Gini) [reply CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Tracking/AnalyticsScreen.swift:11]: I will remove it.

**Review by zladzeyka — APPROVED** [review]

**Pull request merged**